python/ngen_conf/src/ngen/config/init_config/cfe.py

```diff
@@ -0,0 +1,165 @@
+
+
+m = Literal["m"]
+m_per_m = Literal["m/m"]
+empty = Literal[""]
+
+
+class CFEBase(serde.IniSerializerDeserializer):
+    forcing_file: Literal["BMI"] = "BMI"
+    # soil depth
+    soil_params_depth: FloatUnitPair[m]
+    # beta exponent on Clapp-Hornberger (1978) soil water relations
+    soil_params_b: FloatUnitPair[empty]
+    # saturated hydraulic conductivity
+    soil_params_satdk: FloatUnitPair[Literal["m s-1"]]
+    # saturated capillary head
+    soil_params_satpsi: FloatUnitPair[m]
+    # this factor (0-1) modifies the gradient of the hydraulic head at the soil bottom. 0=no-flow.
+    soil_params_slop: FloatUnitPair[m_per_m]
+    # saturated soil moisture content
+    soil_params_smcmax: FloatUnitPair[m_per_m]
+    # wilting point soil moisture content
+    soil_params_wltsmc: FloatUnitPair[m_per_m]
+    soil_params_expon: FloatUnitPair[empty] = FloatUnitPair(value=1.0, unit="")
+    soil_params_expon_secondary: FloatUnitPair[empty] = FloatUnitPair(
+        value=1.0, unit=""
+    )
+    # maximum storage in the conceptual reservoir
+    max_gw_storage: FloatUnitPair[m]
+    # the primary outlet coefficient
+    cgw: FloatUnitPair[Literal["m h-1"]]
+    # exponent parameter (1.0 for linear reservoir)
+    expon: FloatUnitPair[empty]
+    # initial condition for groundwater reservoir - it is the ground water as a decimal fraction of
+    # the maximum groundwater storage (max_gw_storage) for the initial timestep
+    gw_storage: FloatUnitPair[m_per_m]  # 50%
+    # field capacity
+    alpha_fc: FloatUnitPair[empty]
+    # initial condition for soil reservoir - it is the water in the soil as a decimal fraction of
+    # maximum soil water storage (smcmax * depth) for the initial timestep
+    soil_storage: FloatUnitPair[m_per_m]  # 66.7%
+    # number of Nash lf reservoirs (optional, defaults to 2, ignored if storage values present)
+    k_nash: FloatUnitPair[empty]
+    # Nash Config param - primary reservoir
+    k_lf: FloatUnitPair[empty]
+    # Nash Config param - secondary reservoir
+    nash_storage: List[float]
+    # Giuh ordinates in dt time steps
+    giuh_ordinates: List[float]
+    # set to `1` if `forcing_file=BMI`
+    num_timesteps: int = Field(1, gte=0)
+
+    # prints various debug and bmi info
+    verbosity: int = Field(0, gte=0, lte=3)
+
+    # direct runoff
+    surface_partitioning_scheme: Literal["Schaake", "Xinanjiang"]
+
+    # TODO: add support for below fields
+
+    # # when `CFE coupled to SoilMoistureProfile`
+    # aet_rootzone: bool  # True, true, 1
+    # # layer of the soil that is the maximum root zone depth. That is, the depth of the layer where the AET is drawn from
+    # max_root_zone_layer: FloatUnitPair[m]
+    # # an array of depths from the surface. Example, soil_layer_depths=0.1,0.4,1.0,2.0
+    # soil_layer_depths: List[float]
+    # # `ice-fraction based runoff` | when `CFE coupled to SoilFreezeThaw`
+    # sft_coupled: bool  # True, true, 1
```

Comment:
@aaraney I know you mentioned you plan to do these, they will be needed when cfe is coupled to SoilMoistureProfiles and SoilFreezeThaw models otherwise not.

Reply:
@ajkhattak, I assume that the presence of these fields in a configuration file indicates to the model that `cfe` is couple to `SoilMoistureProfiles`? Or is there a flag in the config file I can use to determine if these fields should be included or not?

Reply:
You are right.
1. If `aet_rootzone=true` is given in the config file, then `CFE` is coupled to `SoilMoistureProfiles` and two additional parameters need to be provided `soil_layer_depths` (1D array) and `max_root_zone_layer` (double).
2. If `sft_coupled=true` is provided in the config file, then CFE is coupled to `SoilFreezeThaw` model, now if the `surface_partitioning_scheme=Schaake`, then one additional parameter `ice_content_threshold` is needed in the config file. If `surface_partitioning_scheme=Xinanjiang` then no additional parameters are needed.

Let me know if you have any questions...